### ISSUE TITLE
Function invocation syntax questions

### ISSUE THREAD

**Issue by romix** [richardhundt/shine#30]
I'm playing a bit with chaining of function invocations, both using the usual function invocation style and OOP invocation style (using dot). I also play with passing parameters to functions without providing them inside brackets, i.e. using `f x` instead of 'f(x)'.  Here are some observations and proposals for your consideration.

Let's say we have a class:

```
class Point
   self(x, y)
      self.x = x
      self.y = y
   end

   move(s)
      self.x += s
      return self
   end
end
-- An object of this class
p = Point(11, 22)
```
1. Now we want to move `p` twice. Please have a look at the following examples. Pay attention to the space characters.

`p. move (3) . move(4)` does not work.
`p . move (3) . move (4)` does not work.
`p . move(3) . move (4)` does work (no space between move and 3)
`p . move(3) . move 4` does work (no space between move and 3, no brackets around 4)
`p . move 3 . move 4` does not work
`p . move 3` does work

May be it is the expected behavior, but it seems to me that Nyanga's parser is way too sensitive when it comes to spaces and other delimiters like tabs, etc.
1. `p move 3` does not work (dot is absent). But it could be actually very nice to be able to invoke a method on an object without writing  a dot (e.g. Scala allows for it). It results e.g. in a more fluent style. What do you think of it?
2. I'd also like to have chaining like this:
   `p move 3 move 4`

```
function two(x, y)
    print x, y
    return two
end
```

`two(1, 2)(3, 4)` works
`two (1, 2) (3, 4)` does not work
`two 1, 2` works
`two 1 2` does not work (no comma). But it could be nice to be able to do so.
`two 1,2 3,4` does not work (because `2 3`is probably considered to be an invocation. Could be nice if it could be understood as `two(1,2)(3,4)`

To summarize:
It could be interesting to be able to use a bit less explicit syntax for functions/method invocations, if possible. I.e. less dots, commas, brackets. But I can also imagine that it is not easy/possible, because of lack of static typing or something like this.

Would be interesting to hear what you think about it...
